### PR TITLE
RR-847 - Combine `reasons_not_to_work` and `affect_ability_to_work`

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR is the 2nd of the API changes to support the Question Set changes. The target branch for this PR is our integration branch `RR-810-integration`

---

This PR implements the API changes to combine the `reasons_not_to_work` and `affect_ability_to_work` questions (in order to support UI ticket RR-818)

Essentially the fields `notHopingToWorkReasons` and `notHopingToWorkOtherReason` have been removed from all the various classes, and the enums `AffectAbilityToWork` (at the API, domain and persistence layers) have had their values changed to reflect the merged set of values.

The flyway script migrates the existing `reasons_not_to_work` records into `affect_abiity_to_work`, renames some existing values (some of the enum values have changed names but semantically are the same thing), and then drops the old table / columns.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/induction/WorkOnReleaseEntity.kt [hunk 1]
```diff
@@ -117,23 +108,16 @@ enum class HopingToWork {
   NOT_SURE,
 }
 
-enum class NotHopingToWorkReason {
-  LIMIT_THEIR_ABILITY,
-  FULL_TIME_CARER,
+enum class AffectAbilityToWork {
```
Comment (nathanrussell-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/induction/WorkOnReleaseEntity.kt:111]: As well as merging the 2 enums, we are renaming a few (correcting the spelling of "offence" 😁 , and making the previous "health", "health_issues" and "no_reason" more verbose and explicit)
The rename of existing values in the database is covered in the flyway script below 👇

File: src/main/resources/db/migration/common/V2024.06.12.0001__affect_ability_to_work.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+UPDATE affecting_ability_to_work
+    SET affect = 'REFUSED_SUPPORT_WITH_NO_REASON'
+    WHERE affect = 'NO_REASON';
+
+--- 4. Drop the table `not_working_reasons`
+DROP TABLE not_working_reasons;
+
+--- 5. Drop the column `work_on_release.not_hoping_other`
+ALTER TABLE work_on_release
+    DROP COLUMN not_hoping_other;
```
Comment (james-baker-digital-justice) [inline src/main/resources/db/migration/common/V2024.06.12.0001__affect_ability_to_work.sql:51]: Can we make the remaining column not null?

Comment (nathanrussell-moj-digital) [reply src/main/resources/db/migration/common/V2024.06.12.0001__affect_ability_to_work.sql:51]: I'm not sure I follow.
This flyway does not rename any columns, and the column that I think you're referring to (`affecting_ability_to_work.affect` ?) is already has a not-null constraint as part of it's original table creation DDL ('V2023.10.17.0001__add_induction_related_tables.sql'):
```
-- Collection Table affecting_ability_to_work
CREATE TABLE affecting_ability_to_work
(
    work_on_release_id UUID NOT NULL,
    affect VARCHAR(50) NOT NULL
);
```

**Review by james-baker-digital-justice — APPROVED** [review]
LGTM

**Pull request merged**